pkg/engine/imageVerify.go

```diff
@@ -125,12 +123,12 @@ type imageVerifier struct {
 	resp          *response.EngineResponse
 }
 
-func (iv *imageVerifier) verify(imageVerify *v1.ImageVerification, images map[string]context.ImageInfo) {
+func (iv *imageVerifier) verify(imageVerify *v1.ImageVerification, images map[string]imageutils.ImageInfo) {
 	imagePattern := imageVerify.Image
 
-	for _, imageInfo := range images {
+	for path, imageInfo := range images {
 		image := imageInfo.String()
-		jmespath := utils.JsonPointerToJMESPath(imageInfo.JSONPointer)
+		jmespath := utils.JsonPointerToJMESPath(path)
 		changed, err := iv.policyContext.JSONContext.HasChanged(jmespath)
```

Comment:
This specific logic also looks weird to me. The image signatures/attestations might change independently to the changes in image digest (an image with a fixed digest might gain new signatures or have existing signatures removed). I am not quite sure why we skip checks if the image was not changed. @JimBugwadia do you have any context behind this?

I would prefer if we removed this specific change and verified the image each time the resource has been modified.

Reply:
The goal was to skip verifying images when other pod (or custom resource) attributes change. Otherwise, any change to the pod triggers an image verification check which is unnecessary.